### PR TITLE
fix: symbol resolution tree.rs: commutative mul parenthesis

### PR DESCRIPTION
**Pull request by JulienBalianSonos**
Add symbolic resolution of multiplication with parenthesis.

- in case a term is a multiplication itself, flatten it
e.g., (a*b)*c => a*b*c

This is needed to run the Nemo Parakeet encoder (with NNEF asset generated by t2n).

### CODE REVIEW

**Review by kali — CHANGES_REQUESTED** [review]

File: data/src/dim/tree.rs [hunk 1]
```diff
@@ -371,11 +371,23 @@ impl TDim {
                 }
             }
             Mul(terms) => {
+                // in case a term is a multiplication itself, flatten it
+                // e.g., (a*b)*c => a*b*c
+                let mut flattened_terms = vec![];
+                for t in terms {
+                    if let Mul(inner_terms) = t {
```
[the remark this answers is not in the document]
Comment (kali) [reply data/src/dim/tree.rs]: sorry for the double back and forth, i thought about it after the fact.

[the remark this answers is not in the document]
Comment (JulienBalianSonos) [reply data/src/dim/tree.rs]: As discussed offline, `parse_tdim`is indeed simplifying it recursively, while direct notation followed by a single simplify does not. I added a test and the edge case fix.
